Sources/Datadog/Core/DD/CoreLogger.swift

```diff
@@ -0,0 +1,98 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+
+/// The `CoreLogger` provides methods to collect debug information and detect execution errors of the Datadog SDK.
+internal struct CoreLogger: CoreLoggerType {
+    /// The prefix applied to all core logs.
+    private static let prefix = "[DATADOG SDK] 🐶 → "
+
+    /// The date provider for annotating core logs.
+    private let dateProvider: DateProvider
+    /// Formatter used to format the time accordingly for local device.
+    private let dateFormatter: DateFormatterType
+    /// The print function.
+    private let printFunction: (String) -> Void
+    /// V1's verbosity level. Only logs above or equal to this level wil be printed.
+    private let currentVerbosityLevel: () -> LogLevel?
+
+    init(
+        dateProvider: DateProvider,
+        timeZone: TimeZone,
+        printFunction: @escaping (String) -> Void ,
+        verbosityLevel: @escaping () -> LogLevel?
+    ) {
+        self.dateProvider = dateProvider
+        self.dateFormatter = presentationDateFormatter(withTimeZone: timeZone)
+        self.printFunction = printFunction
+        self.currentVerbosityLevel = verbosityLevel
+    }
+
+    func debug(_ message: @autoclosure () -> String, error: Error? = nil) {
```

Comment:
I'm using `@autoclosure` to not evaluate `String` if verbosity is disabled (in production apps). Strings interpolation can be surprisingly expensive in some cases.